lib/3scale/backend/storage_async/client.rb

```diff
@@ -216,6 +101,52 @@ def init_sentinels_client(opts)
 
           Async::Redis::SentinelsClient.new(name, opts[:sentinels], role)
         end
+
+        # Authenticated RESP2 if credentials are provided, RESP2 otherwise
+        def make_redis_protocol(opts)
+          uri = URI(opts[:url] || "")
+          credentials = [ uri.user || opts[:username], uri.password || opts[:password]]
+
+          if credentials.any?
+            AuthenticatedRESP2.new(credentials)
+          else
+            Async::Redis::Protocol::RESP2
+          end
```

Comment:
I think it is better to have a protocol like this and always use it instead of conditionally select a protocol here.

https://github.com/socketry/async-redis/blob/f11a36655aa000fe54c82899631ead0753e6cb37/examples/auth/wrapper.rb#L52-L74

Reply:
Right, I'll try to do this.

Reply:
Done https://github.com/3scale/apisonator/pull/350/commits/a126a9b781b02f040d7743f50c3788319cc23de0

The new protocol also accepts logical dbs: https://github.com/3scale/apisonator/pull/350/commits/821f08cabf8825dd7aa853cefa86d127157d1f21